Source/Particles/SpeciesPhysicalProperties.H

```diff
@@ -0,0 +1,75 @@
+/* Copyright 2020 Maxence Thevenet
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#ifndef WARPX_SPECIESPHYSICALPROPERTIES_H_
+#define WARPX_SPECIESPHYSICALPROPERTIES_H_
+
+#include "WarpXConst.H"
+
+#include <AMReX_REAL.H>
+#include <AMReX_AmrCore.H>
+
+#include <map>
+#include <limits>
+
+enum struct PhysicalSpecies{unspecified=0, electron, positron, photon, hydrogen};
+
+static PhysicalSpecies from_string(std::string species)
+{
+    if( species=="unspecified" )
```

Comment:
I think that, if we envisage to add more options in the future, it would be better to use a map (or an unordered map, as @ax3l has pointed out in the past for a similar case).

Something like:
```
    const auto string2spec = std::map<std::string, PhysicalSpecies>{
        {"unspecified", PhysicalSpecies::unspecified},
        {"electron",PhysicalSpecies::electron},
        {"positron",PhysicalSpecies::positron},
        {"photon",PhysicalSpecies::photon},
        {"hydrogen",PhysicalSpecies::hydrogen}};

    const auto it = string2spec.find(species);
    if( it != string2spec.end()){
        return it->second;
    }
    else{
      amrex::Abort("unknown PhysicalSpecies")
    }
```

Reply:
I'll discuss with @ax3l, the initial implementation was with a map indeed :).

Reply:
This is an implementation detail and can be done so, e.g. in the `.cpp` file. I would just refrain from providing this map to the user, we just want this function interface exposed as API.